### DISCUSSION TITLE
Integration of Log4j Honeypot: Log4Pot

### DISCUSSION THREAD

**Discussion by t3chn0m4g3** [telekom-security/tpotce#972] [Announcements]
I am happy to announce the integration of @thomaspatzke's Log4Pot into T-Pot and I hope it will help you on your Log4j hunting endeavors.

If you setup a new instance of T-Pot there is not much to do. Simply choose the `LOG4J` flavor.
If you are already using a T-Pot 20.06 you have to run the `/opt/tpot/update.sh` script, [import](https://github.com/telekom-security/tpotce/wiki/Import---Export-Kibana-Objects) the new objects into Kibana and run the `/opt/tpot/bin/tped.sh` script and choose `log4j`.

![image](https://user-images.githubusercontent.com/4318452/146459419-88bd8f56-46f3-4405-b7d2-caf127482d07.png)

**Comment by certrik** [comment]
Hi. Great stuff @t3chn0m4g3 
Is there a proposed path to upgrade the log4pot while it evolves?
Is ...
```
--download-payloads: Download http(s) and ldap payloads and log indicators
--download-class: Implement downloading Java Class file referenced by the payload
--download-dir: Set a download directory. If given, payloads are stored persistently and are not deleted after analysis
```
... also integrated wit log4pot@TPOTCE?

Thank you very much.

**Comment by t3chn0m4g3** [comment]
Yes.

**Comment by t3chn0m4g3** [comment]
New version of Log4Pot (based on latest master) has been pushed.
You need to run `/opt/tpot/update.sh` for some changes to take effect (automatic payload download etc.).
This adds automatic payload downloads to `/data/log4pot/payloads` and deobfuscation.

**Reply by rahulpawar1209** [reply]
Hi @t3chn0m4g3 ,
I installed updated honeypot version 20.06.2 having Log4Pot but not getting any hits on it. On other honeypots getting lots of hits.

**Reply by archangelneo18** [reply]
same here, updated but not getting any hits for log4pot.

**Reply by t3chn0m4g3** [reply]
It seems there is a problem with a field type preventing the ingestion. Will look into it and provide a fix.
![image](https://user-images.githubusercontent.com/4318452/149787797-88818089-1701-4c41-b302-38afd909987d.png)

**Reply by t3chn0m4g3** [reply]
Just redeployed Logstash, the error should not occur any longer.
You need to import the attached object for the dashboard to show all values (see Wiki on how to import the objects).
[Log4Pot Request.ndjson.zip](https://github.com/telekom-security/tpotce/files/7883344/Log4Pot.Request.ndjson.zip)

**Comment by zarandija** [comment]
Hi.

I have already installed version 20.06 in standard mode (all). If I perform the steps 
"If you are already using a T-Pot 20.06 you have to run the script /opt/tpot/update.sh, import the new objects in Kibana and run the script /opt/tpot/bin/tped.sh and choose log4j", I only see the script log4j and I lose all the other monitors in kibana.

How can I add log4f to the standard script?

Thanks.

![standard vs log4j](https://user-images.githubusercontent.com/80309479/148026433-70215094-01f0-4827-8c59-3c390bc666ad.jpg)

**Reply by t3chn0m4g3** [reply]
You can of course create your own edition by modifying `/opt/tpot/etc/tpot.yml` to your specific use case. You find all available editions in `/opt/tpot/etc/compose/`.

**Reply by zarandija** [reply]
I will give it a try. Thanks for the reply.

**Reply by eschultze** [reply]
Hello, I tried to merge nextgen.yml and log4j.yml by running `docker-compose -f nextgen.yml -f log4j.yml config` into a new.yml but it didn't work. Also manually adding the log4j_local network and the whole segment of log4j into a new file went very badly as seen below.

![image](https://user-images.githubusercontent.com/12034548/148100282-f8b2fea0-d5f5-45a8-8b87-a94af5ae3e40.png)

Let me know if you pull it off @zarandija

**Comment by zarandija** [comment]
Hi
After 60 days of use, when I go to see the statistics with Kibana,https://1.2.3.4:64297/kibana/app/home#/
It shows me the "Welcome Home" page and in the T-Pot Dashboard, all without data and only indicates.
"1 of 48 shards failed
The data you are seeing might be incomplete or wrong."
can it be solved?
![image](https://user-images.githubusercontent.com/80309479/149464710-f5f0779e-c6e9-4a46-b7c9-82055a9d46ed.png)

Thnks

**Reply by t3chn0m4g3** [reply]
That usually means that at some point there was an unclean shutdown of the machine and at least one of the daily indices broke at that point. You can open the KIbana => Stack Management => Index Management and look for the ones which are `red`. The quickest solution is usually to delete them.

**Reply by zarandija** [reply]
Hi t3chn0m4g3

It is working again. Really it was that the hard disk was full, sorry, 200GB is valid for 40 days in my case.

Regards.
![image](https://user-images.githubusercontent.com/80309479/149722678-87ba32a0-274c-46c3-bb0a-fe000e5fc3b7.png)

**Reply by t3chn0m4g3** [reply]
Awesome! With the next major release I will be introducing the Index Lifecycle Management and as such, the retention policy is directly attached to the daily index with possible changes right from within Kibana.

This should make it a lot easier in the future to adjust for personal needs.

![image](https://user-images.githubusercontent.com/4318452/149745191-a300fcd5-2beb-4dc8-aff9-14242e935533.png)
![image](https://user-images.githubusercontent.com/4318452/149745391-724bc10e-e4cf-4f5b-87c0-aff7fb1ddd31.png)

**Comment by thomaspatzke** [comment]
Log4Pot got a lot of updates in the last weeks. I've also renamed log4pot.py to log4pot-server.py to resolve a naming collision issue with the Python module.

**Reply by t3chn0m4g3** [reply]
Great, wasn't able to build it and will update it.

**Reply by t3chn0m4g3** [reply]
Just deployed Log4Pot based on the latest master. Changes are looking very good!
Thank you @thomaspatzke and community 😃
